lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -164,6 +164,7 @@ def add_amount(doc, money, options)
       def add_personal_info(doc, credit_card, options)
         doc.send('first-name', credit_card.first_name)
         doc.send('last-name', credit_card.last_name)
+        doc.send('personal-identification-number') if options[:personal_identification_number]
```

Comment:
Good lesson for something to keep in mind for tests. Remote tests don't check if an optional value actually reaches the gateway (and the gateway doesn't care because it's optional so the response doesn't help), and since we mock responses in Unit tests, it doesn't actually verify the request either, unless using `.check_request`. Of course, that level of test doesn't seem necessary for basic fields like this, so just something to watch out for.

Reply:
😞Good catch @molbrown. I had copy/pasted `doc.email` and moved on to testing. 

@curiousepic Good to know - thanks for the info!